Source/Core/DolphinQt/Debugger/CodeDiffProxyModel.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <optional>
+
+#include <QSortFilterProxyModel>
+#include <QString>
+
+#include "Common/CommonTypes.h"
+
+namespace Core
+{
+class BranchWatch;
+}
+class CodeDiffDialog;
+class CodeDiffTableModel;
+class QGridLayout;
+
+class CodeDiffProxyModel final : public QSortFilterProxyModel
+{
+  Q_OBJECT
+
+  friend CodeDiffDialog;
+
+public:
+  explicit CodeDiffProxyModel(const Core::BranchWatch& branch_watch, QObject* parent = nullptr)
+      : QSortFilterProxyModel(parent), m_branch_watch(branch_watch)
+  {
+  }
+  CodeDiffTableModel* sourceModel() const;
+  bool filterAcceptsRow(int source_row, const QModelIndex& source_parent) const override;
+
+  template <bool CodeDiffProxyModel::*member>
+  void OnToggled(bool enabled)
+  {
+    this->*member = enabled;
+    invalidateRowsFilter();
+  }
+  void OnSymbolTextChanged(const QString& text)
+  {
+    m_symbol_name = text;
+    invalidateRowsFilter();
+  }
+  template <std::optional<u32> CodeDiffProxyModel::*member>
+  void OnAddressTextChanged(const QString& text)
+  {
+    bool ok;
+    if (const u32 value = text.toUInt(&ok, 16); ok)
+      this->*member = value;
+    else
+      this->*member = std::nullopt;
+    invalidateRowsFilter();
+  }
+  void OnDelete(const QModelIndex& index);
+  void OnDelete(QModelIndexList index_list);
+
+  bool IsBranchFiltered(u32 hex) const;
+
+private:
+  const Core::BranchWatch& m_branch_watch;
+
+  QString m_symbol_name;
+  std::optional<u32> m_origin_min, m_origin_max, m_destin_min, m_destin_max;
+  bool m_b, m_bl, m_bc, m_bcl, m_blr, m_blrl, m_bclr, m_bclrl, m_bctr, m_bctrl, m_bcctr, m_bcctrl;
```

Comment:
These should have initial default values for deterministic initialization. Also maybe put some of these on new lines, this is quite a bit in sequence (ditto for the optionals).

Reply:
- The std::optional members will always be std::nullopt when default constructed.
- The bool members are initialized in `CodeDiffDialog::CreateLayout()` when the QCheckBoxes are connected and checked (in that order).
- In retrospect, I think the default construction of the QString member explains why I needed to check `QString::isNull()` in `CodeDiffProxyModel::filterAcceptsRow`.  I will go ahead and give the QString a default value.

Reply:
I'm saying they should be defaulted so on construction they have a valid state. Which is good because it prevents weird uninitialized/undefined behavior from happening if refactors happen, and for whatever reason, any of them are accidentally touched before the call is made (this is a reasonable request).

I know the optional members default to a default valid state. I was more referring to line breaking, rather than initializing those.